packages/react/react/src/components/OutsideClick/OutsideClick.tsx

```diff
@@ -0,0 +1,51 @@
+import { ComponentProps, ElementType, forwardRef, ReactNode, useState } from 'react';
+import { useOutsideClickEffect } from '../../index';
+
+type NonHaveChildElements =
+  | 'input'
+  | 'textarea'
+  | 'img'
+  | 'br'
+  | 'hr'
+  | 'meta'
+  | 'link'
+  | 'base'
+  | 'col'
+  | 'embed'
+  | 'source'
+  | 'track'
+  | 'wbr';
+
+type NoChildren<Tag extends ElementType> = Tag extends NonHaveChildElements
+  ? { children?: never }
+  : { children: ReactNode };
+
+type TagRequired<Tag extends ElementType> = Tag extends 'div' ? { as?: Tag } : { as: Tag };
+
+type AllowedTagName<Tag extends ElementType> = Tag extends keyof HTMLElementTagNameMap
+  ? HTMLElementTagNameMap[Tag]
+  : HTMLElement;
+
+type OutsideClickProp<Tag extends ElementType> = ComponentProps<Tag> &
+  TagRequired<Tag> &
+  NoChildren<Tag> & {
+    callback: () => void;
+  };
+
+function OutsideClick<Tag extends ElementType = 'div'>({ as, children, callback, ...props }: OutsideClickProp<Tag>) {
```

Comment:
If I go over `ref` from props, will ref throw it away without using the value?

Reply:
That was the part I missed! 🥲
Thank you for your review! ☺️

I solved using existing [`@toss/react` `useCombinedRefs`](https://github.com/toss/slash/blob/0a9a89e0dd6a3d8f86264d0ec4bb062d075337ea/packages/react/react/src/hooks/useCombinedRefs.ts#L5) !

I'd appreciate it if you could check it out! 🙌